src/vs/base/browser/ui/highlightedlabel/highlightedLabel.ts

```diff
@@ -44,14 +44,51 @@ export class HighlightedLabel implements IDisposable {
 			highlights = [];
 		}
 
+		const originalHighlights = [...highlights];
 		this.text = text;
 		this.highlights = highlights;
+		if (escapedNewLines) {
+			this.highlights = this.adjustHighlightsForEscapedLineBreaks(this.highlights);
+		}
 		this.render();
+		this.highlights = originalHighlights;
```

Comment:
Why is `highlights` being reset? If we decide to escape newlines shouldn't the field respect that?